open_spiel/python/examples/lewis_signaling_qlearner.py

```diff
@@ -0,0 +1,297 @@
+        payoffs_str = ",".join([str(x) for x in payoffs.flatten()])
+    else:
+        payoffs_str = FLAGS.payoffs
+        try:
+            payoffs_list = [float(x) for x in payoffs_str.split(",")]
+            payoffs = np.array(payoffs_list).reshape((num_states, num_states))
+        except ValueError:
+            raise ValueError(
+                'There should be {} (states * actions) elements in payoff. Found {} elements'
+                .format(num_states * num_states, len(payoffs_list)))
+
+    env_configs = {
+        "num_states": num_states,
+        "num_messages": num_messages,
+        "payoffs": payoffs_str
+    }
+
+    env = rl_environment.Environment(game, **env_configs)
+
+    if FLAGS.compare:
+        rewards_list = []
+        opts_list = []
+        converge_point_list = []
+        percent_opt_list = []
+        for centralized in [True, False]:
+            rewards, opts, converge_point, percent_opt = run_experiment(
+                num_players, env, payoffs, centralized)
+            rewards_list += [rewards]
+            opts_list += [opts]
+            converge_point_list += [converge_point]
+            percent_opt_list += [percent_opt]
+    else:
+        rewards, opts, converge_point, percent_opt = run_experiment(
+            num_players, env, payoffs, FLAGS.centralized)
+        rewards_list = [rewards]
+        opts_list = [opts]
+        converge_point_list = [converge_point]
+        percent_opt_list = [percent_opt]
+
+    if FLAGS.plot:
+        import matplotlib as mpl
+        import matplotlib.pyplot as plt
+        import scipy.stats as stats
+
+        params = {
+            "font.size": 12,
+            "axes.labelsize": 12,
+            "xtick.labelsize": 11,
+            "ytick.labelsize": 11,
+        }
+        mpl.rcParams.update(params)
+
+        def init_fig():
+            fig, ax = plt.subplots(1, 1)
+            ax.spines["top"].set_visible(False)
+            ax.spines["right"].set_visible(False)
+            return fig, ax
+
+        def plot_scalars(scalars,
+                         repetition_axis=0,
+                         scalar_labels=None,
+                         title=None,
+                         ax_labels=None):
+            """
+            Plots scalar on ax by filling 1 standard error
+            Args:
+                scalars: List of scalars to plot (mean taken over repetition axis)
+                repetition_axis: Axis to take the mean over
+                scalar_labels: Labels for the scalars (for legend)
+                title: Figure title
+                ax_labels: Labels for x and y axis (list of 2 strings)
+            """
+            assert all([len(s.shape) == 2 for s in scalars
+                       ]), "Only 2D arrays supported for plotting"
+
+            if scalar_labels is None:
+                scalar_labels = [None] * len(scalars)
+
+            assert len(scalars) == len(
```

Comment:
use assertEqual

Reply:
Google code style recommended using exceptions for argument checking. I will convert the assert statements to exceptions unless assertEqual is better.